src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs

```diff
@@ -0,0 +1,256 @@
+        /// Initialize a ScheduleView element
+        /// </summary>
+        private void InitScheduleView(Autodesk.Revit.DB.ViewSchedule view)
+        {
+            InternalSetScheduleView(view);
+        }
+
+        /// <summary>
+        ///     Initialize a ViewSchedule Element.
+        /// </summary>
+        /// <param name="category"></param>
+        /// <param name="name"></param>
+        /// <param name="type"></param>
+        private void InitScheduleView(Category category, string name, ScheduleType type)
+        {
+            Autodesk.Revit.DB.Document doc = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(doc);
+
+            Autodesk.Revit.DB.ViewSchedule vs = CreateViewSchedule(category, name, type);
+
+            InternalSetScheduleView(vs);
+
+            TransactionManager.Instance.TransactionTaskDone();
+
+            ElementBinder.CleanupAndSetElementForTrace(doc, this.InternalElement);
+        }
+
+        #endregion
+
+        #region Private mutators
+
+        /// <summary>
+        /// Set the InternalViewSchedule property and the associated element id and unique id
+        /// </summary>
+        /// <param name="view"></param>
+        private void InternalSetScheduleView(Autodesk.Revit.DB.ViewSchedule view)
+        {
+            this.InternalViewSchedule = view;
+            this.InternalElementId = view.Id;
+            this.InternalUniqueId = view.UniqueId;
+        }
+
+        #endregion
+
+        #region Private helper methods
+
+        private static Autodesk.Revit.DB.ViewSchedule CreateViewSchedule(Category category, string name, ScheduleType type)
+        {
+            Autodesk.Revit.DB.Document doc = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(doc);
+
+            Autodesk.Revit.DB.ViewSchedule viewSchedule = null;
+            switch (type)
+            {
+                case ScheduleType.KeySchedule:
+                    viewSchedule = Autodesk.Revit.DB.ViewSchedule.CreateKeySchedule(doc, new Autodesk.Revit.DB.ElementId(category.Id));
+                    viewSchedule.Name = name;
+                    break;
+            }
+
+            TransactionManager.Instance.TransactionTaskDone();
+
+            return viewSchedule;
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        ///     Create Schedule by Category, Type and Name.
+        /// </summary>
+        /// <param name="category"></param>
+        /// <param name="name"></param>
+        /// <param name="scheduleType"></param>
+        /// <returns name="scheduleView">Schedule</returns>
+        public static ScheduleView CreateSchedule(Category category, string name, string scheduleType)
+        {
+            if (category == null || name == null)
+            {
+                throw new System.ArgumentNullException("Category or Name");
```

Comment:
Please throw the `ArgumentNullException` individually with a message like "You must supply a Category." That message should be added as a string resource to the Resources.resx and Resources.en-US.resx files. We use these files for localizing for different countries.